src/Compression/CompressedReadBufferBase.cpp

```diff
@@ -42,15 +43,24 @@ static void validateChecksum(char * data, size_t size, const Checksum expected_c
     if (expected_checksum == calculated_checksum)
         return;
 
-    std::stringstream message;
+    WriteBufferFromOwnString message;
 
     /// TODO mess up of endianness in error message.
     message << "Checksum doesn't match: corrupted data."
         " Reference: " + getHexUIntLowercase(expected_checksum.first) + getHexUIntLowercase(expected_checksum.second)
         + ". Actual: " + getHexUIntLowercase(calculated_checksum.first) + getHexUIntLowercase(calculated_checksum.second)
         + ". Size of compressed block: " + toString(size);
 
-    const char * message_hardware_failure = "This is most likely due to hardware failure. If you receive broken data over network and the error does not repeat every time, this can be caused by bad RAM on network interface controller or bad controller itself or bad RAM on network switches or bad CPU on network switches (look at the logs on related network switches; note that TCP checksums don't help) or bad RAM on host (look at dmesg or kern.log for enormous amount of EDAC errors, ECC-related reports, Machine Check Exceptions, mcelog; note that ECC memory can fail if the number of errors is huge) or bad CPU on host. If you read data from disk, this can be caused by disk bit rott. This exception protects ClickHouse from data corruption due to hardware failures.";
+    const char * message_hardware_failure = "This is most likely due to hardware failure. "
```

Comment:
That is worse readable on github, but maybe better for ide

Reply:
This line was too long, now it looks much better in ide